### PR TITLE
Book/inventory unit tests

### PR DESCRIPTION
**Pull request by j-ctrl7**
Changes to Book, Book Repo, sort feature, and ORM diagram update

### CODE REVIEW

File: src/main/java/com/example/bookstore/book/BookService.java [hunk 1]
```diff
@@ -27,6 +27,25 @@ public Iterable<Book> getBookByQuery(String query) {
...
+            }
+            case "publisher" -> {
+                return this.bookRepository.findAllByOrderByPublisherAsc();
+            }
+            case "stock" -> {
+                return this.bookRepository.findAllByOrderByStockAsc();
+            }
+        }
+        // somehow invalid sort param passed
+        return null;
```
Comment (chase-scott) [inline src/main/java/com/example/bookstore/book/BookService.java:46]: Should we be throwing an exception here instead?

**Review by chase-scott — APPROVED** [review]
Very nice work.

**Review by Favourolotu — APPROVED** [review]
Excellent work

File: src/main/java/com/example/bookstore/book/BookService.java (hunk 1 above)
Comment (j-ctrl7) [reply src/main/java/com/example/bookstore/book/BookService.java:46]: not sure. I think null works as well

**Pull request merged**